### PR TITLE
Feature/full site render

### PR DESCRIPTION
**Pull request by abegong**
This branch implements full-site rendering.

* Adds breadcrumbs to individual data asset pages
* Creates a new index.html page

https://www.loom.com/share/4d17b1ba554145cc9a9e43257f279227

Note:
* index.html currently renders its own, special one-time only document type through a special jinja renderer. This is a janky. I will deprecate as soon as humanly possible.

### CODE REVIEW

File: great_expectations/cli/cli.py [hunk 1]
```diff
@@ -163,6 +163,7 @@ def init(target_directory):
         logger.critical(err.message)
         sys.exit(-1)
 
+    # TODO: There should be a method in the DataContext to get this directory.
```
Comment (jcampbell) [inline great_expectations/cli/cli.py]: I think you're looking for:
```
    @property
    def root_directory(self):
```

File: great_expectations/data_context/data_context.py [hunk 2]
```diff
@@ -1452,6 +1456,8 @@ def render_full_static_site(self):
             None
         """
 
+        index_links = []
+
         # TODO: this is a temporary implementation and should be replaced with a rendered specific for this purpose
```
Comment (jcampbell) [inline great_expectations/data_context/data_context.py]: Ready to remove this caveat, yes?

File: great_expectations/data_context/data_context.py [hunk 3]
```diff
@@ -1463,9 +1469,34 @@ def render_full_static_site(self):
             model = DescriptivePageRenderer.render(validation)
             out_filepath = self.get_validation_doc_filepath(data_asset_name, expectation_suite_name)
             safe_mmkdir(os.path.dirname(out_filepath))
+            print(out_filepath)
```
Comment (jcampbell) [inline great_expectations/data_context/data_context.py]: drop?

File: great_expectations/render/view/templates/navbar.j2 [hunk 4]
```diff
@@ -1,7 +1,7 @@
 <nav id="navigation" class="navbar navbar-light bg-light sticky">
     <nav class="nav nav-pills flex-column">
         {% if data_asset_name -%}
-            <a class="navbar-brand" href="#">{{data_asset_name}}</a>
+            <a class="navbar-brand" href="#">{{data_asset_name[2]}}</a>
```
Comment (jcampbell) [inline great_expectations/render/view/templates/navbar.j2]: hard-coded index?

File: tests/conftest.py [hunk 5]
```diff
@@ -14,7 +14,7 @@
 
 from .test_utils import get_dataset
 
-CONTEXTS = ['PandasDataset', 'sqlite', 'SparkDFDataset']
+CONTEXTS = ['PandasDataset', 'sqlite']#, 'SparkDFDataset']
```
Comment (jcampbell) [inline tests/conftest.py]: please revert before PR

File: tests/conftest.py [hunk 6]
```diff
@@ -157,6 +157,12 @@ def sqlitedb_engine():
     return sa.create_engine('sqlite://')
 
 
+@pytest.fixture()
```
Comment (jcampbell) [inline tests/conftest.py]: ???

File: tests/test_cli.py [hunk 7]
```diff
@@ -269,3 +270,13 @@ def test_cli_profile(empty_data_context, filesystem_csv_2, capsys):
     assert "Profiling 'my_datasource' with 'BasicDatasetProfiler'" in captured.out
     assert "Note: You will need to review and revise Expectations before using them in production." in captured.out
     logger.removeHandler(handler)
+
+
+def test_scaffold_directories_and_notebooks(empty_directory):
```
Comment (jcampbell) [inline tests/test_cli.py]: I think just use tmp_path_factory as the fixture here

File: tests/data_context/test_data_context.py [hunk 8]
```diff
@@ -397,3 +399,133 @@ def test_data_context_result_store(titanic_data_context):
         data_asset_name = profiling_result[1]['meta']['data_asset_name']
         validation_result = titanic_data_context.get_validation_result(data_asset_name, "BasicDatasetProfiler")
         assert data_asset_name in validation_result["meta"]["data_asset_name"]
+
+def test_render_full_static_site(tmp_path_factory, filesystem_csv_3):
```
Comment (jcampbell) [inline tests/data_context/test_data_context.py:407]: Please see the new smoketest pattern in tests/render/test_render.py

**Review by jcampbell — CHANGES_REQUESTED** [review]

File: great_expectations/cli/cli.py (hunk 1 above)
Comment (abegong) [reply great_expectations/cli/cli.py]: 🙇

File: great_expectations/data_context/data_context.py (hunk 3 above)
Comment (abegong) [reply great_expectations/data_context/data_context.py]: 👍

File: great_expectations/render/view/templates/navbar.j2 (hunk 4 above)
Comment (abegong) [reply great_expectations/render/view/templates/navbar.j2]: Assuming data_asset_names are always triples, this isn't such a crazy assumption.

I'd like to spend a little more time thinking about the best way to set up site navigation---but not today.

File: tests/conftest.py (hunk 5 above)
Comment (abegong) [reply tests/conftest.py]: My mistake.

Fixing now

File: tests/test_cli.py (hunk 7 above)
Comment (abegong) [reply tests/test_cli.py]: :+1:

**Pull request merged**